SplunkAppForWazuh/appserver/static/js/services/sourceTypeStorageService/sourceTypeStorageService.js

```diff
@@ -0,0 +1,36 @@
+define(['../module'], function(app) {
+    'use strict'
+
+    class SourceTypeStorageService {
+        constructor() {
+            this.sessionStorage = sessionStorage
+        }
+
+        /**
+         * Removes the selected SourceType
+         */
+        removeSourceType() {
+            delete this.sessionStorage.selectedSourceType
+        }
+
+        /**
+         * Select an SourceType by name
+         * @param {String} sourceType
+         */
+        setSourceType(sourceType) {
+            this.sessionStorage.selectedSourceType = `{"sourceType":"${sourceType}"}`
+        }
+
+        /**
+         * Returns currently selected SourceType
+         * @param {String} sourceType
+         */
+        getSourceType() {
+            if (this.sessionStorage.selectedSourceType) {
+                return JSON.parse(this.sessionStorage.selectedSourceType)
```

Comment:
is it possible that the JSON.parse fails? should a possible exception be handled?

Reply:
Yes, as every expression is possible to fail, the use of try and catch seems correct to me. Thanks a lot for the suggestion.